testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/javascript/JavascriptAdapterTest.java

```diff
@@ -400,6 +397,69 @@ public void loginRequiredAction() {
                 .init(defaultArguments(), this::assertSuccessfullyLoggedIn);
     }
 
+    /**
+     * Test for scope handling via explicit Keycloak configuration: <pre>{@code
+     * new Keycloak({.... scope: "profile email phone"})
+     * }</pre>
+     * See KEYCLOAK-14412
+     */
+    @Test
+    public void testOnLoadRequireLoginShouldHonorScopes() {
+
+        JSObjectBuilder config = JSObjectBuilder.create()
+                .add("url", authServerContextRootPage + "/auth")
+                .add("realm", REALM_NAME)
+                .add("clientId", CLIENT_ID)
+                .add("scope", "profile email phone") // phone is optional client scope
+                ;
+
+        JSObjectBuilder initOptions = defaultArguments().loginRequiredOnLoad();
+
+        try {
+            testExecutor.configure(config)
+                        .init(initOptions);
+            // This throws exception because when JavascriptExecutor waits for AsyncScript to finish
+            // it is redirected to login page and executor gets no response
+
+            throw new RuntimeException("Probably the login-required OnLoad mode doesn't work, because testExecutor should fail with error that page was redirected.");
+        } catch (WebDriverException ex) {
+            // should happen
+        }
+
+        testExecutor.loginForm(testUser, this::assertOnTestAppUrl)
+                .init(initOptions, this::assertSuccessfullyLoggedIn)
+        .executeAsyncScript("var callback = arguments[arguments.length - 1];" +
```

Comment:
This doesn't need to be async script, it is used if we need an output from asynchronous function, in this case you can use simple return like this:
```suggestion
       .executeScript("return window.keycloak.tokenParsed.scope", assertOutputContains("phone"));
```